src/authentication/authentication-hapi-plugin.ts

```diff
@@ -106,9 +94,37 @@ class AuthenticationHapiPlugin extends HapiPlugin {
 
   public async getTeamTokenHandler(request: Hapi.Request, reply: Hapi.IReply) {
     try {
-      const teamId = parseInt(request.paramsArray[0], 10);
-      const teamToken: TeamToken[] = await teamTokenQuery(this.db, undefined, teamId);
-      if (!teamToken.length) {
+      const credentials = request.auth.credentials as AccessToken;
+      const userName = sanitizeUsername(credentials.sub);
+      const isAdmin = await this.isAdmin(userName);
+      const userTeams = await this.gitlab.getUserGroups(userName);
+      const parameterKey = 'teamIdOrName';
+      const teamIdOrName = request.params[parameterKey];
+      let ownTeam: Group | undefined;
+      if (userTeams && teamIdOrName) {
+        ownTeam = userTeams.find(assertTeam.bind(undefined, teamIdOrName));
```

Comment:
Maybe it's good to discuss this more generally. I feel that `bind` is basically language level support for currying, albeit with a bit dicky syntax. When writing the function, I don't need to think about explicitly returning functions, which gets cumbersome as the arity increases.

```js
const addA = (x, y) => x + y;
const incrementA = addA.bind(undefined, 1); // dicky syntax
const resultA = addA(2, 3); 

const addB = x => y => x + y; // arguably weird looking definition
const incrementB = addB(1);
const resultB = addB(2)(3); // dicky syntax
```

In this case I don't feel strongly about this really, if the function is never applied with all the arguments directly. However, I'd suggest that in general it's ok to curry with `bind`.